### PR TITLE
CRA teardown, webpack setup with babel, and react --> closes #15

### LINKED ISSUE #15: Teardown project and set it up again with webpack (don't use create react app)
Create a branch that tears down the existing project and rebuilds it using webpack. We need to do this in order to use CSS modules so issue #13 depends on this (is blocked by this).

### PR DESCRIPTION
**Pull request by armanhk**
The following are the intended accomplishments of this PR:

- [x] Wipe codebase setup with CRA
- [x] Set up repo from scratch with Webpack, Babel, and React
- [x] Pull in CSS from firebase spike #11 
- [x] Ensure all necessary loaders are present
- [x] Ensure react module renders equivalently (plus some styling adjustments) to pre-CRA-teardown

**Note**:
Initially went back and forth on the notion of doing the work to port to CSS modules (#13) with this ticket. Ultimately decided not to do so although  [ae1f420](https://github.com/armanhk/Module-React-Login/pull/16/commits/ae1f420dc2e149d43dcd1e8c2896dc5344cabef1) contains some CSS modules setup items which were removed in subsequent commits.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,31 +1,38 @@
...
+  "description": "",
+  "main": "index.js",
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "start": "webpack-dev-server --open --hot --mode development",
+    "build": "webpack --mode production",
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (yessenbayev) [inline package.json]: no tests

File: package.json [hunk 2]
```diff
@@ -1,31 +1,38 @@
...
-      "not dead",
-      "not op_mini all"
-    ],
-    "development": [
-      "last 1 chrome version",
-      "last 1 firefox version",
-      "last 1 safari version"
-    ]
+  "author": "armanhk, yessenbayev",
+  "license": "ISC",
```
Comment (yessenbayev) [inline package.json]: Set the license to GNU

File: src/App.js [hunk 3]
```diff
@@ -1,36 +1,59 @@
...
-    console.log(username,password);
+    delegate(username, password);
   }
 
-  render(){
-      const element =
-      <div id="login-modal">
-        <form id="login-form" onSubmit={this.submitRoutine}>
+  // Renders the login UI
+  renderLogin() {
```
Comment (yessenbayev) [inline src/App.js]: These code changes are outside of the scope of #15

File: src/App.js [hunk 4]
```diff
@@ -1,36 +1,59 @@
...
             <hr className="login-divider"></hr>
             <button className="login-submit login-button" type="submit">Log In</button>
-            <button className="login-button" onClick={(e)=>{e.preventDefault()}}>Sign Up</button>
+            <button className="login-button" onClick={(e)=>{this.submitRoutine(e, this.props.signupDelegate)}}>Sign Up</button>
         </form> 
       </div>;
+  }
+
+  // Renders the logout button (UI for logged in state for demo)
+  renderLogout() {
```
Comment (yessenbayev) [inline src/App.js]: These code changes are outside of the scope of #15

File: src/App.js [hunk 5]
```diff
@@ -1,36 +1,59 @@
...
+    return <div id="login-modal">
+        <h1 className="logout-header">Hello, {this.props.userEmail}</h1>
+        <button className="logout-button" onClick={this.props.logoutDelegate}>Log Out</button>
+      </div>
+  }
+
+  // Component render function
+  render(){
```
Comment (yessenbayev) [inline src/App.js]: These code changes are outside of the scope of #15

File: src/App.module.css [hunk 6]
```diff
@@ -0,0 +1,174 @@
+html, body {
```
Comment (yessenbayev) [inline src/App.module.css]: These code changes are outside of the scope of #15. Put the style changes into #17.

File: src/index.html [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html class="no-js">
```
Comment (yessenbayev) [inline src/index.html]: Why is the class "no-js" here?

File: src/index.html [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+<head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <link rel="icon" href="./src/favicon.ico" type="image/x-icon">
+    <title>Webpack Setup</title>
```
Comment (yessenbayev) [inline src/index.html]: Incorrect Title for the webpage, please use the title present in the master branch

File: src/index.js [hunk 9]
```diff
@@ -1,11 +1,70 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import './index.css';
 import LoginComponent from './App';
-import './serviceWorker.js';
 
-ReactDOM.render(<LoginComponent />, document.getElementById('root'));
+/**
```
Comment (yessenbayev) [inline src/index.js]: Firebase authentification is outside of scope of #15. If used for testing, please indicate properly.

File: webpack.config.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        rules: [
+            {
+                test: /\.(js|jsx)$/,
+                exclude: /node_modules/,
+                use: {
+                    loader: "babel-loader"    
+                }
+            },
+            {
+                test: /\.html$/,
```
Comment (yessenbayev) [inline webpack.config.js]: Add "exclude: /node_modules/," or take our node_modules from the local scope.

File: webpack.config.js [hunk 11]
```diff
@@ -20,7 +20,19 @@ module.exports = {
...
+                    {
+                        loader: "css-loader",
+                        options: {
+                            modules: {
+                                localIdentName: "[local]---[hash:base64:5]"
+                            },
+                            importLoaders: 1
+                        }
+                    },
+                    "postcss-loader"
```
Comment (yessenbayev) [inline webpack.config.js]: Nit: personally, I like to add commas after each element of a sequential data structure.

File: src/App.module.css [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+    /* Outline tested with colorblind filters provided by Colorblinding Chrome extension */
+    outline: solid rgba(255, 0, 0, 0.432);
+}
+
+.login-button:hover {
+    color: rgb(44, 44, 46);
+    background: rgb(97, 219, 251);
+}
+
+.logout-header {
```
Comment (yessenbayev) [inline src/App.module.css]: Make sure that there is no dead code.

File: package.json (hunk 2 above)
Comment (armanhk) [reply package.json]: de459f25c2796b7036d349c205fa860ca0d81691

File: src/App.js (hunk 3 above)
Comment (armanhk) [reply src/App.js]: de459f25c2796b7036d349c205fa860ca0d81691

File: src/App.js (hunk 4 above)
Comment (armanhk) [reply src/App.js]: de459f25c2796b7036d349c205fa860ca0d81691

File: src/App.js (hunk 5 above)
Comment (armanhk) [reply src/App.js]: de459f25c2796b7036d349c205fa860ca0d81691

File: webpack.config.js (hunk 10 above)
Comment (armanhk) [reply webpack.config.js]: de459f25c2796b7036d349c205fa860ca0d81691

File: webpack.config.js (hunk 11 above)
Comment (armanhk) [reply webpack.config.js]: de459f25c2796b7036d349c205fa860ca0d81691

File: src/App.module.css (hunk 12 above)
Comment (armanhk) [reply src/App.module.css]: de459f25c2796b7036d349c205fa860ca0d81691

File: package.json (hunk 1 above)
Comment (armanhk) [reply package.json]: https://github.com/armanhk/Module-React-Login/issues/18

File: src/index.html (hunk 7 above)
Comment (armanhk) [reply src/index.html]: No change, module index.html used for testing only
